datamodel_code_generator/model/base.py

```diff
@@ -234,8 +234,40 @@ def get_template(template_file_path: Path) -> Template:
 
 def get_module_path(name: str, file_path: Optional[Path]) -> List[str]:
     if file_path:
+        # Filter out temporary directory paths and only use meaningful module components
```

Comment:
This is necessary since for lithic-schemagen changes, we need to create better paths so that our polymorphic/discriminator references aren't messed up. This is a patch on top of the logic we have, to make everything work once more (this is necessary to run it as is, even without polymorphic references)